src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -1458,11 +1470,13 @@ private void ShowInvocationExamples()
 
             List<ITemplateInfo> templateList = _matchedTemplates.Select(x => x.Info).ToList();
             Reporter.Output.WriteLine("Examples:");
+            HashSet<string> usedGroupIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
             foreach (string preferredName in preferredNameList)
             {
                 ITemplateInfo template = templateList.FirstOrDefault(x => string.Equals(x.ShortName, preferredName, StringComparison.OrdinalIgnoreCase));
-                if (template != null)
+                string identity = string.IsNullOrWhiteSpace(template.GroupIdentity) ? string.IsNullOrWhiteSpace(template.Identity) ? string.Empty : template.Identity : template.GroupIdentity;
```

Comment:
I think this has similar problems to the previous, except with blank template identities. I was thinking something like:

if (template != null && 
    (string.IsNullOrEmpty(template.GroupIdentity) || usedGroupIds.Add(identity)))